apps/gallery/js/ImageEditor.js

```diff
@@ -644,13 +644,15 @@ function exitEdit(saved) {
   // we want the slider re-zeroed for the next time we come into the editor
   exposureSlider.forceSetExposure(0.0);
 
-  // We came in to edit mode from fullscreenView.  If the user cancels the edit
-  // go back to fullscreenView.  Otherwise, if the user saves the photo, we go
-  // back to thumbnail list view because that is where the newly saved
-  // image is going to show up.
-  // XXX: this isn't really right. Ideally the new photo should show up
-  // right next to the old one and we should go back to fullscreenView to view
-  // the edited photo.
+  // If exitEdit is called without saved argument e.g. when device storage
+  // is unavailable, exit now and do not check for saved.
+  if (arguments.length < 1) {
```

Comment:
That's correct exitEdit passes values saved for the two cases inside ImageEditor.js